### ISSUE TITLE
Build error of version 1.0.0

### ISSUE THREAD

**Issue by oseiberts11** [timonwong/uwsgi_exporter#43]
If I build version 1.0.0 in container "golang:alpine" which presumably has a recent version of go, it fails to build, like so:
```
$ make precheck style lint build
make: Nothing to be done for 'precheck'.
>> checking code style
mkdir -p /go/bin
curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/v1.18.0/install.sh \
	| sed -e '/install -d/d' \
	| sh -s -- -b /go/bin v1.18.0
golangci/golangci-lint info checking GitHub for tag 'v1.18.0'
golangci/golangci-lint info found version: 1.18.0 for v1.18.0/linux/amd64
golangci/golangci-lint info installed /go/bin/golangci-lint
>> running golangci-lint
GO111MODULE=on go list -e -compiled -test=true -export=false -deps=true -find=false -tags= -- ./... > /dev/null
go: downloading github.com/prometheus/client_golang v1.0.0
go: downloading github.com/prometheus/common v0.7.0
go: downloading github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: downloading github.com/stretchr/testify v1.3.0
go: downloading gopkg.in/alecthomas/kingpin.v2 v2.2.6
go: downloading github.com/beorn7/perks v1.0.0
go: downloading github.com/golang/protobuf v1.3.2
go: downloading github.com/prometheus/procfs v0.0.2
go: downloading github.com/sirupsen/logrus v1.4.2
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751
go: downloading github.com/alecthomas/units v0.0.0-20190717042225-c3de453c63f4
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading golang.org/x/sys v0.0.0-20190422165155-953cdadca894
GO111MODULE=on /go/bin/golangci-lint run  ./...
uwsgi_exporter.go:1: /usr/local/go/src/sort/slice.go:18:14: undeclared name: any (typecheck)
package main
exporter/reader_unix_test.go:10:18: cannot use t (variable of type *testing.T) as assert.TestingT value in argument to assert.New: wrong type for method Errorf (typecheck)
	a := assert.New(t)
	                ^
exporter/exporter_test.go:126:16: cannot use t (variable of type *testing.T) as assert.TestingT value in argument to assert.Equal: wrong type for method Errorf (typecheck)
		assert.Equal(t, expect, got, "Wrong main stats")
		             ^
make: *** [Makefile.common:158: common-lint] Error 1
```
Avoiding lint and doing a direct `make build` on my Mac also fails (differently):
```
~/git/uwsgi_exporter$ make build
>> building binaries
GO111MODULE=on /Users/oseibert/go/bin/promu build --prefix /Users/oseibert/git/uwsgi_exporter
 >   uwsgi_exporter
# golang.org/x/sys/unix
vendor/golang.org/x/sys/unix/zsyscall_darwin_amd64.go:28:3: //go:linkname must refer to declared function or variable
vendor/golang.org/x/sys/unix/zsyscall_darwin_amd64.go:43:3: //go:linkname must refer to declared function or variable
vendor/golang.org/x/sys/unix/zsyscall_darwin_amd64.go:59:3: //go:linkname must refer to declared function or variable
vendor/golang.org/x/sys/unix/zsyscall_darwin_amd64.go:75:3: //go:linkname must refer to declared function or variable
vendor/golang.org/x/sys/unix/zsyscall_darwin_amd64.go:90:3: //go:linkname must refer to declared function or variable
vendor/golang.org/x/sys/unix/zsyscall_darwin_amd64.go:105:3: //go:linkname must refer to declared function or variable
vendor/golang.org/x/sys/unix/zsyscall_darwin_amd64.go:121:3: //go:linkname must refer to declared function or variable
vendor/golang.org/x/sys/unix/zsyscall_darwin_amd64.go:136:3: //go:linkname must refer to declared function or variable
vendor/golang.org/x/sys/unix/zsyscall_darwin_amd64.go:151:3: //go:linkname must refer to declared function or variable
vendor/golang.org/x/sys/unix/zsyscall_darwin_amd64.go:166:3: //go:linkname must refer to declared function or variable
vendor/golang.org/x/sys/unix/zsyscall_darwin_amd64.go:166:3: too many errors
!! command failed: build -o /Users/oseibert/git/uwsgi_exporter/uwsgi_exporter -ldflags -X github.com/prometheus/common/version.Version=1.0.0 -X github.com/prometheus/common/version.Revision=9f88775cc1a600e4038bb1eae5edfdf38f023dc4 -X github.com/prometheus/common/version.Branch=master -X github.com/prometheus/common/version.BuildUser=oseibert@oseibert -X github.com/prometheus/common/version.BuildDate=20220810-15:56:20  -mod=vendor -a -tags 'netgo' github.com/timonwong/uwsgi_exporter: exit status 2
make: *** [common-build] Error 1
```
This is with `go version go1.18.3 darwin/amd64`.

**Comment by Rhialto** [comment]
If I try on a different machine with go 1.18.3, I get this:
```
murthe:.../scratch/tmp/uwsgi_exporter$ git reset --hard
HEAD is now at 9f88775 Adjust CHANGELOG
murthe:.../scratch/tmp/uwsgi_exporter$ gmake
>> checking code style
>> running check for unused/missing packages in go.mod
GO111MODULE=on go mod tidy
>> running check for unused packages in vendor/
GO111MODULE=on go mod vendor
diff --git a/go.sum b/go.sum
index 5f7b59c..1943f39 100644
--- a/go.sum
+++ b/go.sum
@@ -1,15 +1,12 @@
-github.com/alecthomas/template v0.0.0-20160405071501-a0175ee3bccc h1:cAKDfWh5VpdgMhJosfJnn5/FoN2SRZ4p7fJNX58YPaU=
 github.com/alecthomas/template v0.0.0-20160405071501-a0175ee3bccc/go.mod h1:LOuyumcjzFXgccqObfd/Ljyb9UuFJ6TxHnclSeseNhc=
 github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751 h1:JYp7IbQjafoB+tBA3gMyHYHrpOtNuDiK/uB5uXxq5wM=
 github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751/go.mod h1:LOuyumcjzFXgccqObfd/Ljyb9UuFJ6TxHnclSeseNhc=
-github.com/alecthomas/units v0.0.0-20151022065526-2efee857e7cf h1:qet1QNfXsQxTZqLG4oE62mJzwPIB8+Tee4RNCL9ulrY=
 github.com/alecthomas/units v0.0.0-20151022065526-2efee857e7cf/go.mod h1:ybxpYRFXyAe+OPACYpWeL0wqObRcbAqCMya13uyzqw0=
 github.com/alecthomas/units v0.0.0-20190717042225-c3de453c63f4 h1:Hs82Z41s6SdL1CELW+XaDYmOH4hkBN4/N9og/AsOv7E=
 github.com/alecthomas/units v0.0.0-20190717042225-c3de453c63f4/go.mod h1:ybxpYRFXyAe+OPACYpWeL0wqObRcbAqCMya13uyzqw0=
 github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973/go.mod h1:Dwedo/Wpr24TaqPxmxbtue+5NUziq4I4S80YR8gNf3Q=
 github.com/beorn7/perks v1.0.0 h1:HWo1m869IqiPhD389kmkxeTalrjNbbJTC8LXupb+sl0=
 github.com/beorn7/perks v1.0.0/go.mod h1:KWe93zE9D1o94FZ5RNwFwVgaQK1VOXiVxmqh+CedLV8=
-github.com/davecgh/go-spew v1.1.0 h1:ZDRjVQ15GmhC3fiQ8ni8+OwkZQO4DARzQgrnXU1Liz8=
 github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=
 github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
gmake: *** [Makefile.common:182: common-unused] Error 1
```
which, to my eye which is not really trained in go, looks bad (because it stops). However, if I try `make -k` to continue anyway:
```
murthe:.../scratch/tmp/uwsgi_exporter$ gmake -k
>> checking code style
>> running check for unused/missing packages in go.mod
GO111MODULE=on go mod tidy
>> running check for unused packages in vendor/
GO111MODULE=on go mod vendor
diff --git a/go.sum b/go.sum
index 5f7b59c..1943f39 100644
--- a/go.sum
+++ b/go.sum
@@ -1,15 +1,12 @@
-github.com/alecthomas/template v0.0.0-20160405071501-a0175ee3bccc h1:cAKDfWh5VpdgMhJosfJnn5/FoN2SRZ4p7fJNX58YPaU=
 github.com/alecthomas/template v0.0.0-20160405071501-a0175ee3bccc/go.mod h1:LOuyumcjzFXgccqObfd/Ljyb9UuFJ6TxHnclSeseNhc=
 github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751 h1:JYp7IbQjafoB+tBA3gMyHYHrpOtNuDiK/uB5uXxq5wM=
 github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751/go.mod h1:LOuyumcjzFXgccqObfd/Ljyb9UuFJ6TxHnclSeseNhc=
-github.com/alecthomas/units v0.0.0-20151022065526-2efee857e7cf h1:qet1QNfXsQxTZqLG4oE62mJzwPIB8+Tee4RNCL9ulrY=
 github.com/alecthomas/units v0.0.0-20151022065526-2efee857e7cf/go.mod h1:ybxpYRFXyAe+OPACYpWeL0wqObRcbAqCMya13uyzqw0=
 github.com/alecthomas/units v0.0.0-20190717042225-c3de453c63f4 h1:Hs82Z41s6SdL1CELW+XaDYmOH4hkBN4/N9og/AsOv7E=
 github.com/alecthomas/units v0.0.0-20190717042225-c3de453c63f4/go.mod h1:ybxpYRFXyAe+OPACYpWeL0wqObRcbAqCMya13uyzqw0=
 github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973/go.mod h1:Dwedo/Wpr24TaqPxmxbtue+5NUziq4I4S80YR8gNf3Q=
 github.com/beorn7/perks v1.0.0 h1:HWo1m869IqiPhD389kmkxeTalrjNbbJTC8LXupb+sl0=
 github.com/beorn7/perks v1.0.0/go.mod h1:KWe93zE9D1o94FZ5RNwFwVgaQK1VOXiVxmqh+CedLV8=
-github.com/davecgh/go-spew v1.1.0 h1:ZDRjVQ15GmhC3fiQ8ni8+OwkZQO4DARzQgrnXU1Liz8=
 github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=
 github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
gmake: *** [Makefile.common:182: common-unused] Error 1
>> building binaries
GO111MODULE=on /home/rhialto/go/bin/promu build --prefix /mnt/scratch/scratch/tmp/uwsgi_exporter 
 >   uwsgi_exporter
>> running all tests
GO111MODULE=on go test   -mod=vendor ./...
?       github.com/timonwong/uwsgi_exporter     [no test files]
ok      github.com/timonwong/uwsgi_exporter/exporter    (cached)
gmake: Target 'common-all' not remade because of errors.
```
This seems to succeed. I'm puzzled...

**Comment by timonwong** [comment]
Thanks for reporting, the code repo is somehow out of date. I'll fix that later

**Comment by timonwong** [comment]
Addressed in #44
